shared/modules/selectors.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  ALLOWED_SMART_TRANSACTIONS_CHAIN_IDS,
+  SKIP_STX_RPC_URL_CHECK_CHAIN_IDS,
+} from '../constants/smartTransactions';
+import { ENVIRONMENT } from '../../development/build/constants';
+import {
+  getCurrentChainId,
+  getCurrentNetwork,
+  accountSupportsSmartTx,
+} from '../../ui/selectors/selectors'; // TODO: Migrate shared selectors to this file.
+import { getNetworkNameByChainId } from './feature-flags';
+
+export const getSmartTransactionsOptInStatus = (
+  state: Record<string, any>,
+): boolean | null => {
+  return state.metamask.preferences?.stxOptIn;
+};
+
+export const getIsAllowedStxChainId = (state: Record<string, any>): boolean => {
+  const chainId = getCurrentChainId(state);
+  return ALLOWED_SMART_TRANSACTIONS_CHAIN_IDS.includes(chainId);
+};
+
+const getIsAllowedRpcUrlForStx = (state: Record<string, any>) => {
+  const chainId = getCurrentChainId(state);
+  const isNotDevelopment =
+    process.env.METAMASK_ENVIRONMENT !== ENVIRONMENT.DEVELOPMENT &&
+    process.env.METAMASK_ENVIRONMENT !== ENVIRONMENT.TESTING;
+  if (!isNotDevelopment || SKIP_STX_RPC_URL_CHECK_CHAIN_IDS.includes(chainId)) {
```

Comment:
Minor, but we're negating a boolean called `isNot`, would it be clearer to set `isDevelopment` and use that?